sys-info-base.ts

```diff
@@ -208,38 +208,37 @@ export class SysInfoBase implements ISysInfo {
 
 	private exec(cmd: string, execOptions?: IExecOptions): string | any {
 		try {
-			if(cmd) {
+			if (cmd) {
 				return this.$childProcess.exec(cmd, null, execOptions).wait();
 			}
-		} catch(e) {
+		} catch (e) {
 			// if we got an error, assume not working
 		}
 
 		return null;
 	}
 
-	// `android -h` returns exit code 1 on successful invocation (Mac OS X for now, possibly Linux). Therefore, we cannot use $childProcess
-	private checkAndroid(pathToAndroid: string): IFuture<boolean> {
-		return ((): boolean => {
+	private checkEmulator(): IFuture<boolean> {
+		return (() => {
 			let result = false;
 			try {
-				if(pathToAndroid) {
-					let androidChildProcess = this.$childProcess.spawnFromEvent(pathToAndroid, ["-h"], "close", {}, {throwError: false}).wait();
-					result = androidChildProcess && androidChildProcess.stdout && _.includes(androidChildProcess.stdout, "android");
-				}
-			} catch(err) {
-				this.$logger.trace(`Error while checking is ${pathToAndroid} installed. Error is: ${err.messge}`);
+				// emulator -help exits with code 1 on Windows, so we should parse the output.
+				// First line of it should be:
+				// Android Emulator usage: emulator [options] [-qemu args]
+				const emulatorHelp = this.$childProcess.spawnFromEvent(this.$androidEmulatorServices.pathToEmulatorExecutable, ["-help"], "close", {}, { throwError: false }).wait();
+				result = emulatorHelp.stdout.indexOf("usage: emulator") !== -1;
+			} catch (err) {
+				this.$logger.trace(`Error while checking is emulator installed. Error is: ${err.messge}`);
```

Comment:
I think we will never trace the error message from `emulator -help`, if there is one, because we set `throwError: false` on spawnFromEvent. You can check if `result` is `false` and trace the message from `emulatorHelp.stdout` or `emulatorHelp.stderr`.